core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java

```diff
@@ -112,14 +112,12 @@ public Patient convert(JSONObject patientJSON)
             }
 
             Set<Feature> features = this.convertFeatures(patientJSON);
-            Set<Disorder> disorders = this.convertDisorders(patientJSON, this.disorderTerm,
-                MIM_VOCABULARY, ApiConfiguration.JSON_DISORDERS);
-            Set<Disorder> clinicalDisorders = this.convertDisorders(patientJSON, this.clinicalDisorderTerm,
-                ORDO_VOCABULARY, ApiConfiguration.JSON_DIAGNOSIS);
+            Set<Disorder> disorders = null;
+            Set<Disorder> clinicalDisorders = null;
+            this.convertDisorders(patientJSON, disorders, clinicalDisorders);
             Set<MatchingPatientGene> genes = this.convertGenes(patientJSON);
 
-            if ((features == null || features.isEmpty()) &&
-                (genes == null || genes.isEmpty())) {
+            if ((features == null || features.isEmpty()) && (genes == null || genes.isEmpty())) {
```

Comment:
Unrelated whitespace change

Reply:
better codestyle, && should be either on new line or inline with code

Reply:
To make reviewing easier, whitespace/style fixes should be in separate commits from functional changes

Reply:
sure, just allowed this small one here :)